src/lib/reservationTimes.ts

```diff
@@ -0,0 +1,131 @@
+import type { SettingsStore } from '$lib/settings';
+import { ReservationCategory } from '$types';
+import * as dtu from './datetimeUtils';
+
+export const minPoolStart = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getMinPoolStartTime(date));
+export const maxPoolEnd = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getMaxPoolEndTime(date));
+export const minClassroomStart = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getMinClassroomStartTime(date));
+export const maxClassroomEnd = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getMaxClassroomEndTime(date));
+export const resCutoff = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getReservationCutOffTime(date));
+export const cancelCutoff = (
+	stns: SettingsStore,
+	cat: ReservationCategory,
+	date: string
+): number => {
+	if ([ReservationCategory.pool, ReservationCategory.classroom].includes(cat)) {
+		return 0;
+	} else {
+		return dtu.timeStrToMin(stns.getCancelationCutOffTime(date));
+	}
+};
+export const inc = (stns: SettingsStore, date: string): number =>
+	dtu.timeStrToMin(stns.getReservationIncrement(date));
+
+export const minuteOfDay = (date: Date): number => date.getHours() * 60 + date.getMinutes();
+
+export function validReservationDate(
+	stns: SettingsStore,
+	date: Date,
+	category: ReservationCategory
+) {
+	return dtu.datetimeToLocalDateStr(date) >= minValidDateStr(stns, category);
+}
+
+export function beforeResCutoff(
+	stns: SettingsStore,
+	dateStr: string,
+	startTime: string,
+	category: ReservationCategory
+): boolean {
+	let now = dtu.PanglaoDate();
+	let tomorrow = dtu.PanglaoDate();
+	tomorrow.setDate(now.getDate() + 1);
+	let tomStr = dtu.datetimeToLocalDateStr(tomorrow);
+
+	if ([ReservationCategory.pool, ReservationCategory.classroom].includes(category)) {
+		return beforeCancelCutoff(stns, dateStr, startTime, category);
+	} else if (dateStr > tomStr) {
+		return true;
+	} else if (dateStr == tomStr && minuteOfDay(now) <= resCutoff(stns, dateStr)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+export function beforeCancelCutoff(
+	stns: SettingsStore,
+	dateStr: string,
+	startTime: string,
+	category: ReservationCategory
+): boolean {
+	let now = dtu.PanglaoDate();
+	let today = dtu.datetimeToLocalDateStr(now);
+	if (dateStr > today) {
```

Comment:
does this actually work? string vs date.

might be better to use `dayjs(dateStr).isAfter(today)`

Reply:
I think they are both strings, no?  

There is a task, Issue #19, for convertring all date ops to use dayjs.  I think we can make that a separate PR?